draft-ietf-ohai-ohttp.md

```diff
@@ -132,44 +132,53 @@ them or linked to other requests.
-  with that key.
+* The details of an HPKE public key that the oblivious encapsulation resource
+  accepts, including an identifier for that key and the HPKE algorithms that
+  are used with that key.
 
-* The identity of an oblivious proxy resource that will forward encapsulated
-  requests and responses to a single oblivious request resource. See {{proxy-state}}
-  for more information about the mapping between oblivious proxy and oblivious
-  request resources.
+* The identity of an oblivious proxy resource that will accept proxy requests
+  carrying an encapsulated request body and forward these request bodies to
+  a single oblivious encapsulation resource. See {{proxy-state}} for more
+  information about the mapping between oblivious proxy and oblivious
+  encapsulation resources.
 
 This information allows the client to make a request of an oblivious target
 resource with that resource having only a limited ability to correlate that
 request with the client IP or other requests that the client might make to that
 server.
 
 ~~~ aasvg
-+---------+       +----------+       +----------+    +----------+
-| Client  |       | Proxy    |       | Request  |    | Target   |
-|         |       | Resource |       | Resource |    | Resource |
-+----+----+       +----+-----+       +----+-----+    +----+-----+
-     |                 |                  |               |
-     | Encapsulated    |                  |               |
-     | Request         |                  |               |
-     +---------------->| Encapsulated     |               |
-     |                 | Request          |               |
-     |                 +----------------->| Request       |
-     |                 |                  +-------------->|
-     |                 |                  |               |
-     |                 |                  |      Response |
-     |                 |     Encapsulated |<--------------+
-     |                 |         Response |               |
-     |    Encapsulated |<-----------------+               |
-     |        Response |                  |               |
-     |<----------------+                  |               |
-     |                 |                  |               |
++---------+      +----------+      +---------------+      +----------+
+| Client  |      | Proxy    |      | Encapsulation |      | Target   |
+|         |      | Resource |      | Resource      |      | Resource |
++----+----+      +----+-----+      +-------+-------+      +----+-----+
+     |                |                    |                   |
+     | Proxy          |                    |                   |
+     | Request        |                    |                   |
+     | [+ Encrypted   |                    |                   |
+     |    Request ]   |                    |                   |
+     +--------------->| Encapsulated       |                   |
+     |                | Request            |                   |
```

Comment:
To clarify, what do you mean when you say it's also encapsulated, and why is that an important detail?

Reply:
It's a word thing.  "Encapsulated" is a label you apply to something that has been put in a capsule.  The capsule and the thing it contains then becomes an encapsulation.  You don't encapsulate the capsule (except recursively, by putting into a larger capsule).

In this case, the request that was encrypted is the thing that has been encapsulated.  The thing you are calling the "Encapsulated Request" is a wrapper or encapsulation, it isn't encapsulated.  The outer request is maybe encapsulating (-ing), but that seems wrong for some reason too.

We generally don't have names for a container and their contents.  Which is what makes this hard.  A briefcase containing papers is just a briefcase.  A bus containing passengers is a bus.  A safe containing a valuable item is a safe.  A bathroom stall is still a bathroom stall, even when it is occupied.  But in all cases, the containing thing is not defined by its contents in quite the same way as we might like here.

Reply:
Maybe instead of Encapsulated Request we call it a Proxied Request? The proxy would then take Proxy Requests as input and output Proxied Requests. 

I would prefer to keep the Encapsulation Resource name for now, as that's at least more precise than Request Resource. 

Thoughts?

Reply:
But all of the requests are proxied, so it isn't very descriptive.  And I don't think that Encapsulation is better (it's not worse) than Request.

Thought: which of the requests are "oblivious"?  Could the outer one be oblivious?

Reply:
Note that the client's request to the proxy in this proposal is a "proxy request," as in, "a request to have something proxied", whereas the proxy's request to the target is a "proxied request," as in, "a request that has _been_ proxied." This distinction seems pretty clear to me.

Reply:
> Thought: which of the requests are "oblivious"?  Could the outer one be oblivious?

The problem with this question is that it's not specific to _which_ request. Is it the client to proxy request or the proxy to target request? I think we need different names for these two things.

Reply:
Resolving based on latest email thread. Thanks for working through this!